File: hora-extra-backend/src/sockets/types/NpcMovePayload.ts
/**
 * Payload para sincronização de NPCs e Bosses.
 */
export interface INpcMovePayload {
    id: string;                  // Identificador Único da Entidade (ex: "guard_01")
    p: [number, number, number]; // Posição [x, y, z]
    r: number;                   // Rotação Y (Yaw)
}

/**
 * Payload para registro inicial de um NPC na sala.
 */
export interface INpcRegisterPayload {
    id: string;
    type: 'npc' | 'boss';
    p: [number, number, number];
    r: number;
    name?: string;
    isMaster?: boolean;
}

File: hora-extra-client/Assets/Scripts/Network/Models/NpcMovePayload.cs
using System;

namespace HoraExtra.Network.Models
{
    /// <summary>
    /// Payload para sincronização de movimento de NPC/Boss.
    /// </summary>
    [Serializable]
    public class NpcMovePayload
    {
        public string id;
        public float[] p; // [x, y, z]
        public float r;   // Rotação Y (Yaw)
    }

    /// <summary>
    /// Payload para registro inicial de NPC na sala.
    /// </summary>
    [Serializable]
    public class NpcRegisterPayload
    {
        public string id;
        public string type;
        public float[] p;
        public float r;
        public string name;
        public bool isMaster;
    }
}
